src/lib/DefaultRuleManager.ts

```diff
@@ -126,6 +133,28 @@ export class DefaultRuleManager implements RuleManager {
 		}
 	}
 
+	protected async emitRunTelemetry(runDescriptorList: ExtendedRunDescriptor[]): Promise<void> {
+		// Telemetry data always specifies an event name.
+		const runTelemetryObject: TelemetryData = {
+			eventName: "ENGINE_EXECUTION"
```

Comment:
telemetry data always has an `eventName` property. Setting it here means it actually has useful information instead of defaulting to `UNKNOWN`.